src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -2638,5 +2638,423 @@ static void verifyArray(TypeSymbol type)
                 Assert.Equal(3, array.Sizes[1]);
             }
         }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_01()
+        {
+            var source = @"
+partial class C
+{
+    public partial int M();
+    public partial long M() => 42; // 1
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,25): error CS8817: Both partial method declarations must have the same return type.
+                //     public partial long M() => 42; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodReturnTypeDifference, "M").WithLocation(5, 25));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_02()
+        {
+            var source = @"
+partial class C
+{
+    public partial long M();
+    public partial int M() => 42; // 1
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,24): error CS8817: Both partial method declarations must have the same return type.
+                //     public partial int M() => 42; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodReturnTypeDifference, "M").WithLocation(5, 24));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_03()
+        {
+            var source = @"
+using System.Collections.Generic;
+#nullable enable
+
+partial class C
+{
+    public partial string? M1();
+    public partial string M1() => ""hello"";
+
+    public partial IEnumerable<string?> M2();
+    public partial IEnumerable<string> M2() => null!;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_04()
+        {
+            var source = @"
+using System.Collections.Generic;
+#nullable enable
+
+partial class C
+{
+    public partial string M1();
+    public partial string? M1() => null; // 1
+
+    public partial IEnumerable<string> M2();
+    public partial IEnumerable<string?> M2() => null!; // 2
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (8,28): error CS8818: Nullability of reference types in return type doesn't match partial method declaration.
```

Comment:
>error [](start = 27, length = 5)

Consider updating comments so it's clear this is a "warning".

Reply:
funny, I guess I created these baselines before filling out some of the boilerplate to make the diagnostic a warning.